### ISSUE TITLE
parsing numbers with exponent fails

### ISSUE THREAD

**Issue by darronschall** [mikechambers/as3corelib#49]
Originally filed by justintalbot on 2008-05-05T20:47:52

What steps will reproduce the problem?
1. trace(JSON.decode("[3.6e-01]"));

What is the expected output? What do you see instead?
Expected: 0.36
Instead: "Unexpected e encountered"

What version of the product are you using? On what operating system?
AS 3 on Windows

Please provide any additional information below.
Browsing the source of JSONTokenizer.as reveals that exponents are not
handled at all.

**Comment by darronschall** [comment]
Updated by mikechambers on 2008-07-02T16:51:55

Original ticket set status to Accepted (we converted to open)

**Comment by darronschall** [comment]
Updated by darron.schall on 2008-10-23T02:07:24

JSONTest::testDecodeScientificNegativeExponent demonstrates that "6.02e-23" is
correctly parsed as a number.  Also, if you look in JSONTokenizer at line 392 (as of
revision 63) you'll see the code to check for scientific notation in numbers.

Rejecting as invalid - make sure you're using the latest from SVN.

Original ticket set status to Invalid (we converted to closed)
